frontend/models/product-list/server.ts

```diff
@@ -428,6 +431,48 @@ function createProductListSection(
    }
 }
 
+type ApiProductListItemOverrides = NonNullable<
+   ApiProductList['itemOverrides']
+>[0];
+
+function formatItemTypeOverrides(
+   itemOverrides: ApiProductListItemOverrides[] | undefined
+): ProductListItemTypeOverrideIndexed {
+   if (!itemOverrides) return {};
+   const convertedOverrides =
+      convertToProductListItemTypeOverrides(itemOverrides);
+   return convertedOverrides.reduce((result, item) => {
+      result[item?.itemType || '*'] = item;
+      return result;
+   }, {} as ProductListItemTypeOverrideIndexed);
+}
+
+function convertToProductListItemTypeOverrides(
+   itemOverrides: ApiProductListItemOverrides[]
+): ProductListItemTypeOverride[] {
+   const formatedOverrides = itemOverrides.map(
+      (itemOverride): ProductListItemTypeOverride | null => {
+         if (
+            itemOverride == null ||
+            itemOverride.__typename !== 'ComponentProductListItemTypeOverride'
+         ) {
+            return null;
+         } else {
+            return {
+               id: itemOverride.id,
+               itemType: itemOverride.itemType ?? null,
```

Comment:
Why are these all `?? null`? I would have expected these fields are always present if blank strings.

Reply:
was trying to keep it consistent with the other types in `types.ts` but we can make them undefinable too.